modules/core/src/main/java/org/apache/ignite/internal/util/HostAndPortRange.java

```diff
@@ -53,60 +53,89 @@ public static HostAndPortRange parse(String addrStr, int dfltPortFrom, int dfltP
 
         String host;
 
+        String portStr;
         int portFrom;
         int portTo;
 
         if (F.isEmpty(addrStr))
             throw createParseError(addrStr, errMsgPrefix, "Address is empty");
 
-        final int colIdx = addrStr.indexOf(':');
+        if (addrStr.contains("[")) { // IPv6
+            int hostEndIdx = addrStr.indexOf(']');
+            if (hostEndIdx == -1) {
+                throw createParseError(addrStr, errMsgPrefix, "IPv6 host is incorrect");
+            }
+            host = addrStr.substring(1, hostEndIdx);
```

Comment:
`substring(1` assumes that `[` is at index `0`, however, the line above `addrStr.contains("[")` allows `[` at any index.

Reply:
Now it checks for `addrStr.charAt(0) == '['`